packages/salesforcedx-vscode-core/src/conflict/timestampConflictDetector.ts

```diff
@@ -51,8 +51,8 @@ export class TimestampConflictDetector {
         component.cacheComponent.fullName
       );
       lastModifiedInCache = cache.getPropertiesForFile(key)?.lastModifiedDate;
-      if (!lastModifiedInCache || lastModifiedInOrg !== lastModifiedInCache) {
```

Comment:
In my investigation, I found that in `TimestampConflictChecker.check()`, there is:
```
        const diffs = detector.createDiffs(result);
        channelService.showCommandWithTimestamp(
          `${nls.localize('channel_end')} ${nls.localize(
            'conflict_detect_execution_name'
          )}\n`
        );
        return await this.handleConflicts(inputs.data, username, diffs);
```

When `createDiffs()` is called, `determineConflicts()` is called, and the conditional here on line 54 was always false, and `diffComponents()` on line 55 was never being called.

Since the diffs array was never being hydrated, later on when when `handleConflicts()` is called (and diffs is passed in), the user would never be prompted.

It seems to me, that `determineConflicts()` shouldn't be using a cache to determine whether the sources should be pushed.  My reasoning for this is, say we do the following:
1. Edit the metadata of an Apex class, and set its apiVersion to say, 51.0
2. Run "deploy source in manifest to org" (and select override)
3. Edit the metadata of the same Apex class, and set its apiVersion to say, 50.0
4. Run "deploy source in manifest to org" (and select override)

At this point, no matter what, when I go back to the org and view the class, it should have the correct apiVersion.  If using a caching mechanism to determine whether to push or not, the second time the file is pushed, the logic would be, "this file is cached, so don't push it", which leads to this bug.

Reply:
This is a great find @jeffb-sfdc. The timestamp based diff detection is a recent change introduced this summer and I think this is an edge case that we missed. You can read more about the design here - https://salesforce.quip.com/yNriA1RWTGUz#UHbACAJxowc. My preference would be instead of removing lastModifiedInCache based checks completely, is there a way to update it correctly when the first deploy happens? It could complicate the existing design, but worth giving it a try.

Reply:
Thanks for the help and insight/background on this the other day @jag-sfdc .  I added the  condition check at line 54 back in, and removed `setPropertiesForFilesDeploy()`, and also updated the unit tests.

Reply:
This looks right to me. Can you please assign @AnanyaJha as the QA for this once you merge? She worked on the initial design and let's make sure we didn't break anything by doing this.